From b97e51c4a06d4893d0f13af17213e966295a2027 Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Tue, 3 Feb 2026 14:02:57 +0100
Subject: [PATCH] ntdll: Rename some builtin unixlib functions to reflect
 current usage.

Unix libs are no longer loaded upon module load.

(cherry picked from commit 5ffc0a61c7e5f43656d9618bc92d9b131179efda)

CW-Bug-Id: #27687
---
 dlls/mmdevapi/main.c           |  2 +-
 dlls/ntdll/unix/loader.c       |  2 +-
 dlls/ntdll/unix/unix_private.h |  2 +-
 dlls/ntdll/unix/virtual.c      | 14 +++++++-------
 dlls/winecrt0/unix_lib.c       |  2 +-
 dlls/wow64/virtual.c           |  6 +++---
 include/winternl.h             |  4 ++--
 7 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/dlls/mmdevapi/main.c b/dlls/mmdevapi/main.c
index d33c1a1d1dc..3a084a7c2fc 100644
--- a/dlls/mmdevapi/main.c
+++ b/dlls/mmdevapi/main.c
@@ -366,7 +366,7 @@ static BOOL load_driver(const WCHAR *name, DriverFuncs *driver)
         return FALSE;
     }
 
-    if ((status = NtQueryVirtualMemory(GetCurrentProcess(), driver->module, MemoryWineUnixFuncs,
+    if ((status = NtQueryVirtualMemory(GetCurrentProcess(), driver->module, MemoryWineLoadUnixLib,
         &driver->module_unixlib, sizeof(driver->module_unixlib), NULL))) {
         ERR("Unable to load UNIX functions: %lx\n", status);
         goto fail;
diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index 2250c489d19..e628f4649d4 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -1738,7 +1738,7 @@ done:
     if (NT_SUCCESS(status) && ext)
     {
         strcpy( ext, ".so" );
-        load_builtin_unixlib( *module, ptr );
+        set_builtin_unixlib_name( *module, ptr );
     }
     free( file );
     return status;
diff --git a/dlls/ntdll/unix/unix_private.h b/dlls/ntdll/unix/unix_private.h
index da2c37de0a8..615fc37455b 100644
--- a/dlls/ntdll/unix/unix_private.h
+++ b/dlls/ntdll/unix/unix_private.h
@@ -335,7 +335,7 @@ extern void virtual_set_large_address_space(void);
 extern void virtual_fill_image_information( const struct pe_image_info *pe_info,
                                             SECTION_IMAGE_INFORMATION *info );
 extern void *get_builtin_so_handle( void *module );
-extern NTSTATUS load_builtin_unixlib( void *module, const char *name );
+extern NTSTATUS set_builtin_unixlib_name( void *module, const char *name );
 extern BOOL is_emulated_code( ULONG_PTR ptr );
 
 extern NTSTATUS get_thread_ldt_entry( HANDLE handle, THREAD_DESCRIPTOR_INFORMATION *info, ULONG len );
diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index c36e3b6854c..319f1bc425c 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -988,9 +988,9 @@ static void load_steam_overlay(const char *unix_lib_path)
 }
 
 /***********************************************************************
- *           get_builtin_unix_funcs
+ *           load_builtin_unixlib
  */
-static NTSTATUS get_builtin_unix_funcs( void *module, BOOL wow, const void **funcs )
+static NTSTATUS load_builtin_unixlib( void *module, BOOL wow, const void **funcs )
 {
     const char *ptr_name = wow ? "__wine_unix_call_wow64_funcs" : "__wine_unix_call_funcs";
     sigset_t sigset;
@@ -1021,9 +1021,9 @@ static NTSTATUS get_builtin_unix_funcs( void *module, BOOL wow, const void **fun
 
 
 /***********************************************************************
- *           load_builtin_unixlib
+ *           set_builtin_unixlib_name
  */
-NTSTATUS load_builtin_unixlib( void *module, const char *name )
+NTSTATUS set_builtin_unixlib_name( void *module, const char *name )
 {
     sigset_t sigset;
     NTSTATUS status = STATUS_SUCCESS;
@@ -6618,15 +6618,15 @@ NTSTATUS WINAPI NtQueryVirtualMemory( HANDLE process, LPCVOID addr,
         case MemoryImageInformation:
             return get_memory_image_info( process, addr, buffer, len, res_len );
 
-        case MemoryWineUnixFuncs:
-        case MemoryWineUnixWow64Funcs:
+        case MemoryWineLoadUnixLib:
+        case MemoryWineLoadUnixLibWow64:
             if (len != sizeof(unixlib_handle_t)) return STATUS_INFO_LENGTH_MISMATCH;
             if (process == GetCurrentProcess())
             {
                 void *module = (void *)addr;
                 const void *funcs = NULL;
 
-                status = get_builtin_unix_funcs( module, info_class == MemoryWineUnixWow64Funcs, &funcs );
+                status = load_builtin_unixlib( module, info_class == MemoryWineLoadUnixLibWow64, &funcs );
                 if (!status) *(unixlib_handle_t *)buffer = (UINT_PTR)funcs;
                 return status;
             }
diff --git a/dlls/winecrt0/unix_lib.c b/dlls/winecrt0/unix_lib.c
index ebf6594e770..e2eb079bd48 100644
--- a/dlls/winecrt0/unix_lib.c
+++ b/dlls/winecrt0/unix_lib.c
@@ -91,6 +91,6 @@ NTSTATUS __attribute__((naked)) __wine_unix_call_arm64ec( unixlib_handle_t handl
 
 NTSTATUS WINAPI __wine_init_unix_call(void)
 {
-    return NtQueryVirtualMemory( GetCurrentProcess(), image_base(), MemoryWineUnixFuncs,
+    return NtQueryVirtualMemory( GetCurrentProcess(), image_base(), MemoryWineLoadUnixLib,
                                  &__wine_unixlib_handle, sizeof(__wine_unixlib_handle), NULL );
 }
diff --git a/dlls/wow64/virtual.c b/dlls/wow64/virtual.c
index cbaf6dd0fa6..fba46280c3e 100644
--- a/dlls/wow64/virtual.c
+++ b/dlls/wow64/virtual.c
@@ -693,11 +693,11 @@ NTSTATUS WINAPI wow64_NtQueryVirtualMemory( UINT *args )
         break;
     }
 
-    case MemoryWineUnixWow64Funcs:
+    case MemoryWineLoadUnixLibWow64:
         return STATUS_INVALID_INFO_CLASS;
 
-    case MemoryWineUnixFuncs:
-        status = NtQueryVirtualMemory( handle, addr, MemoryWineUnixWow64Funcs, ptr, len, &res_len );
+    case MemoryWineLoadUnixLib:
+        status = NtQueryVirtualMemory( handle, addr, MemoryWineLoadUnixLibWow64, ptr, len, &res_len );
         break;
 
     default:
diff --git a/include/winternl.h b/include/winternl.h
index 197ca36f95d..8ef8aa8e9ff 100644
--- a/include/winternl.h
+++ b/include/winternl.h
@@ -2457,8 +2457,8 @@ typedef enum _MEMORY_INFORMATION_CLASS {
     MemoryBadInformationAllProcesses = 13,
     MemoryImageExtensionInformation = 14,
 #ifdef __WINESRC__
-    MemoryWineUnixFuncs = 1000,
-    MemoryWineUnixWow64Funcs,
+    MemoryWineLoadUnixLib = 1000,
+    MemoryWineLoadUnixLibWow64,
 #endif
 } MEMORY_INFORMATION_CLASS;
 
